src/DotNetLightning.Core/Utils/SeqConsumer.fs

```diff
@@ -0,0 +1,58 @@
+namespace DotNetLightning.Utils
+
+type SeqConsumer<'S, 'T> = {
+    Consume: seq<'S> -> Option<seq<'S> * 'T>
+}
+
+type SeqConsumerBuilder<'S>() =
+    member __.Bind<'T0, 'T1>(seqConsumer0: SeqConsumer<'S, 'T0>, func: 'T0 -> SeqConsumer<'S, 'T1>)
+                                : SeqConsumer<'S, 'T1> = {
+        Consume = fun (sequence0: seq<'S>) ->
+            match seqConsumer0.Consume sequence0 with
+            | None -> None
+            | Some (sequence1, value0) ->
+                let seqConsumer1 = func value0
+                seqConsumer1.Consume sequence1
+    }
+
+    member __.Return<'T>(value: 'T)
+                            : SeqConsumer<'S, 'T> = {
+        Consume = fun (sequence: seq<'S>) -> Some (sequence, value)
+    }
+
+    member __.ReturnFrom<'T>(seqConsumer: SeqConsumer<'S, 'T>): SeqConsumer<'S, 'T> =
+        seqConsumer
+
+    member __.Zero(): SeqConsumer<'S, unit> = {
+        Consume = fun (sequence: seq<'S>) -> Some (sequence, ())
+    }
+
+module SeqConsumer =
+    let seqConsumer<'S> = SeqConsumerBuilder<'S>()
```

Comment:
this can be marked private? otherwise should be PascalCase

Reply:
Generally computation expression builders use lowerPascalCase. eg. `seq { .. }`, `async { .. }`, `result { .. }`. This adds `seqConsumer { .. }` as a computation expression.

Reply:
ah right, didn't realize this was the CE name